src/plugins/es_ui_shared/public/request/request.ts

```diff
@@ -36,121 +39,137 @@ export interface UseRequestConfig extends SendRequestConfig {
   deserializer?: (data: any) => any;
 }
 
-export interface UseRequestResponse {
+export interface UseRequestResponse<D = any, E = Error> {
   isInitialRequest: boolean;
   isLoading: boolean;
-  error: null | unknown;
-  data: any;
-  sendRequest: (...args: any[]) => Promise<SendRequestResponse>;
+  error: E | null;
+  data: D | null;
+  sendRequest: (...args: any[]) => Promise<SendRequestResponse<D, E>>;
 }
 
-export const sendRequest = async (
-  httpClient: ng.IHttpService,
-  { path, method, body }: SendRequestConfig
-): Promise<SendRequestResponse> => {
+export const sendRequest = async <D = any, E = Error>(
+  httpClient: HttpSetup,
+  { path, method, body, query }: SendRequestConfig
+): Promise<SendRequestResponse<D, E>> => {
   try {
-    const response = await (httpClient as any)[method](path, body);
-
-    if (typeof response.data === 'undefined') {
-      throw new Error(response.statusText);
-    }
+    const stringifiedBody = typeof body === 'string' ? body : JSON.stringify(body);
+    const response = await httpClient[method](path, { body: stringifiedBody, query });
 
-    return { data: response.data, error: null };
+    return {
+      data: response.data ? response.data : response,
+      error: null,
+    };
   } catch (e) {
     return {
       data: null,
-      error: e.response ? e.response : e,
+      error: e.response && e.response.data ? e.response.data : e.body,
     };
   }
 };
 
-export const useRequest = (
-  httpClient: ng.IHttpService,
+export const useRequest = <D = any, E = Error>(
+  httpClient: HttpSetup,
   {
     path,
     method,
+    query,
     body,
     pollIntervalMs,
     initialData,
     deserializer = (data: any): any => data,
   }: UseRequestConfig
-): UseRequestResponse => {
+): UseRequestResponse<D, E> => {
+  const isMounted = useRef(true);
```

Comment:
I added this to handle the case when the owner component unmounts before a request has resolved.